### DISCUSSION TITLE
ClientResponseError 400: Something went wrong while processing your request.

### DISCUSSION THREAD

**Discussion by ogrotten** [pocketbase/pocketbase#1425] [Q&A] [answered]
```
ClientResponseError 400: Something went wrong while processing your request.
```

This is the error I get when writing to a collection. 

The project is Svelte + Electron. The error occurs both in the display component and in the Electron backend.

Auth has been performed (this is the frontend code) copied directly from the API Preview, and the specified email address belongs to an admin that shows in the control panel

```
export const pb = new PocketBase('http://127.0.0.1:8090')
export const authData = pb.admins.authWithPassword(
	'user@example.com',
	'exampleexample'
)

```

PB log shoes the auth request with http status 200, but what about the request status/validity? "Auth guest"?

![image](https://user-images.githubusercontent.com/8119012/209903160-ec5ea728-6e31-45f8-8821-4fb0bd8e4053.png)

Dev console shows nothing related.

The code I'm using to write is again c/p directly from the API Preview page
```
	const storeFiles = async (incoming) => {
		const data = {
			filename: 'test',
			path: 'test',
			tags_auto: 'RELATION_RECORD_ID',
			tags_manual: 'RELATION_RECORD_ID',
			object: 'JSON',
			field: 'test',
		}

		const record = await pb
			.collection('files')
			.create(data)
			.catch((err) => console.error(`LOG..Filebrowse: err`, err))

		console.log(`conlog: record`, record)
	}
```

This is where it fails, on the assignment to `record`. That catch is what throws the error. I've seen other posts that the console error is ambiguous on purpose, but where is the detail if not in the PB log or the Dev console?

What's going on here?

**Comment by ganigeorgiev** [comment, marked as answer]
400 error code could be thrown because either some of the submitted data fail the collection validations and/or the request client doesn't satisfy the Create API Rule (eg. is not authenticated).

There are several things that you can check:

- Inspect the JS error data, eg. `err.data` (for all available fields please refer to https://github.com/pocketbase/js-sdk#error-handling).

- Inspect the raw error request response in your browser's Inspect tool (if it is client-side request).

- Inspect the 400 request log in the "Admin UI > Logs" for a more detailed message (the screenshot shows the admin auth request not the record create).

- Start PocketBase with `--debug` and see if there is a more detailed log in the console (eg. `./pocketbase serve --debug`).

**Comment by ogrotten** [comment]
Thanks for the response. Point 1 was the info I needed. It basically showed the following

```
tags_auto:
    code: "validation_missing_rel_records"
    message: "Failed to fetch all relation records with the provided ids."
```

The API Preview example `RELATION_RECORD_ID` generates an error. I was going to make a suggestion of some sort related to the series of disconnects that I had here, but I'm not sure how it could have been avoided aside from me being better at ferreting thru the docs. IOW everything I needed to know, in order to fix this problem, is in the docs. That's really great. 

I think however that certain docs could use some links between. Users could probably benefit from the API Preview pages linking back to relevant web docs. For example, the "Create" tab linking back to wherever the "field types" docs are (which only seems to be in the Collections Basics?). Error Handling could also probably use it's own tab in there as well. Even just a top level link to the web docs based on usage. In my example, linking out to the main docs as well as the js-sdk. 

As a completely separate point of feedback, the docs seem to be pretty good. I read in another post here that the more specific errors are hidden as a point of security by obscurity, which makes perfect sense as to why it's buried under `err.data`. However, I would suggest that that explanation be added to the top of the Error Handling doc. It would just . . . help.

All that being said, I would bet these types of things are "mid priority". I have no tangible complaints. The docs seem to be better than a lot of projects and I can appreciate that.

Anyway, now that I better understand where info is, I'll be able to "fish" thru it myself.

Thanks again for the reply.
